### PR TITLE
INT-1205 Refactor for P1X with override

### PR DESCRIPTION
**Pull request by Francois-Gomis**
Override module Zip in Clickup

### CODE REVIEW

**Review by ClaireBonacheraAlma — APPROVED** [review]

File: Model/Adminhtml/Config/FeePlansConfigBackendModel.php [hunk 1]
```diff
@@ -91,12 +91,27 @@ public function _afterLoad(): void
...
+     * @param $key
+     * @param $feePlan
+     * @return bool
+     */
+    public function removeDisallowedPlans($key, $feePlan): bool
+    {
+        if ($key === 'general:1:0:0' || !$feePlan->allowed) {
+            return true;
+        }
+        return false;
```
Comment (syjust-alma) [inline Model/Adminhtml/Config/FeePlansConfigBackendModel.php:115]:
```suggestion
    public function isAllowedPlan($key, $feePlan): bool
    {
        return $key !== 'general:1:0:0 && $feePlan->allowed;
```

File: Model/Adminhtml/Config/FeePlansConfigBackendModel.php [hunk 2]
```diff
@@ -69,7 +69,7 @@ public function beforeSave(): void
         $almaPlans = $this->configHelper->getBaseApiPlansConfig();
         $feePlans = [];
         foreach ($almaPlans as $key => $feePlan) {
-            if ($key === 'general:1:0:0' || !$feePlan->allowed) {
+            if ($this->removeDisallowedPlans($key, $feePlan)) {
```
Comment (syjust-alma) [inline Model/Adminhtml/Config/FeePlansConfigBackendModel.php:72]:
```suggestion
            if (!$this->isAllowedPlan($key, $feePlan)) {
```

File: Model/Adminhtml/Config/FeePlansConfigBackendModel.php [hunk 3]
```diff
@@ -91,12 +91,27 @@ public function _afterLoad(): void
         }
         $feePlans = [];
         foreach ($almaPlans as $key => $feePlan) {
-            if ($key === 'general:1:0:0' || !$feePlan->allowed) {
+            if ($this->removeDisallowedPlans($key, $feePlan)) {
```
Comment (syjust-alma) [inline Model/Adminhtml/Config/FeePlansConfigBackendModel.php:94]:
```suggestion
            if (!$this->isAllowedPlan($key, $feePlan)) {
```

**Review by syjust-alma — APPROVED** [review]

File: Model/Adminhtml/Config/FeePlansConfigBackendModel.php (hunk 3 above)
Comment (Francois-Gomis) [reply Model/Adminhtml/Config/FeePlansConfigBackendModel.php]: done

File: Model/Adminhtml/Config/FeePlansConfigBackendModel.php (hunk 2 above)
Comment (Francois-Gomis) [reply Model/Adminhtml/Config/FeePlansConfigBackendModel.php]: done

**Pull request merged**